### PR TITLE
Enable workload cluster etcd upgrade through controller/gitops

### PR DESCRIPTION
**Pull request by jiayiwang7**
*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by mdsgabriel — APPROVED** [review]

File: controllers/controllers/resource/template.go [hunk 1]
```diff
@@ -106,6 +107,17 @@ func (r *VsphereTemplate) TemplateResources(ctx context.Context, eksaCluster *an
 		}
 	}
 
+	// pin cgroupDriver to systemd for k8s >= 1.21
```
Comment (mrajashree) [inline controllers/controllers/resource/template.go:110]: can we add a comment specifying we won't need this when the controller follows cp first and md second upgrade order?

**Review by mrajashree — APPROVED** [review]
looks great
/

**Pull request merged**